### PR TITLE
Merge allof

### PR DESCRIPTION
**Pull request by tcdsv**

### CODE REVIEW

File: diff/merge_allof_validation_test.go [hunk 1]
```diff
@@ -0,0 +1,1254 @@
+package diff
```
Comment (reuvenharrison) [inline diff/merge_allof_validation_test.go]: Better to put tests in package diff_test

File: diff/merge_allof_unit_test.go [hunk 2]
```diff
@@ -0,0 +1,1281 @@
+package diff
```
Comment (reuvenharrison) [inline diff/merge_allof_unit_test.go]: Better to put tests in package diff_test

I suggest to rename this file to 'merge_allof_test.go'

File: diff/merge_allof.go [hunk 3]
```diff
@@ -0,0 +1,868 @@
...
+	Items                []*openapi3.SchemaRef
+	Required             [][]string
+	Properties           []openapi3.Schemas
+	MinProps             []uint64
+	MaxProps             []*uint64
+	AdditionalProperties []openapi3.AdditionalProperties
+}
+
+// Merge replaces objects under AllOf with a flattened equivalent
+func Merge(schema openapi3.Schema) (*openapi3.Schema, error) {
```
Comment (reuvenharrison) [inline diff/merge_allof.go]: Can we rename this to 'merge' so it's a private function in diff package?

Comment (tcdsv) [reply diff/merge_allof.go]: We can't do that because it's used in diff_test package

File: flatten/merge_allof.go [hunk 4]
```diff
@@ -0,0 +1,867 @@
...
+		schema.AllOf = nil
+		result, err := mergeFields([]*openapi3.Schema{&schema, mergedAllOf})
+		if err != nil {
+			log.Fatal(err.Error())
+		}
+		return result, nil
+	}
+
+	// handle cases where AllOf is nil, but other fields might include AllOf.
+	result, err := handleNestedAllOfCases(&schema)
```
Comment (reuvenharrison) [inline flatten/merge_allof.go]: Note that this will never be called because isListOfObjects will be false

File: flatten/merge_allof.go [hunk 5]
```diff
@@ -0,0 +1,867 @@
...
+		if s != "" {
+			result = append(result, s)
+		}
+	}
+
+	return result
+}
+
+func isListOfObjects(schema *openapi3.Schema) bool {
+	if schema == nil || schema.AllOf == nil {
```
Comment (reuvenharrison) [inline flatten/merge_allof.go]: I suggest to remove this function since there is only one simple test here

**Comment by reuvenharrison** [comment]
Hi @tcdsv and all,
The [last commit](https://github.com/Tufin/oasdiff/pull/374/commits/0a7316f47193bbbefd783434d886064e546e6cc2) simplifies the way diff uses of Merge.
Instead of merging schemas during the diff, we flatten all specs before running the diff.
Please let me know if you have any questions or comments.
Reuven

**Pull request merged**